src/lib/modernExtend.ts

```diff
@@ -102,12 +120,217 @@ export function setupConfigureForReporting(
+                    msg.data.batteryAlarmState & 1<<22 ||
+                    msg.data.batteryAlarmState & 1<<23
+                ) > 0;
+                if (args.lowStatus) payload.battery_low = battery1Low || battery2Low || battery3Low;
+            }
+
+            return payload;
+        },
+    }];
+
+    const toZigbee: Tz.Converter[] = [
+        {
+            key: ['battery', 'voltage'],
+            convertGet: async (entity, key, meta) => {
+                try {
+                    // Don't fail GET reqest if reading this attribute fails
+                    await entity.read('genPowerCfg', ['batteryPercentageRemaining', 'batteryVoltage']);
+                } catch (e) {
+                    meta.logger.debug(`Reading attribute failed: ${e}, device probably doesn't support it`);
+                }
+            },
+        },
+    ];
+
+    const result: ModernExtend = {exposes, fromZigbee, toZigbee, isModernExtend: true};
+
+    const defaultReporting: ReportingConfigWithoutAttribute = {min: '1_HOUR', max: 'MAX', change: 10};
+    if (args.percentageReporting || args.voltageReporting) {
+        result.configure = async (device, coordinatorEndpoint, logger) => {
+            if (args.percentageReporting) {
+                await setupAttributes(device, coordinatorEndpoint, 'genPowerCfg', [
+                    {attribute: 'batteryPercentageRemaining', ...(args.percentageReportingConfig ?? defaultReporting)},
+                ], logger);
+            }
+            if (args.voltageReporting) {
+                await setupAttributes(device, coordinatorEndpoint, 'genPowerCfg', [
+                    {attribute: 'batteryVoltage', ...(args.voltageReportingConfig ?? defaultReporting)},
+                ], logger);
+            }
+        };
+    }
+
+    if (args.voltageToPercentage || args.dontDividePercentage) {
+        const meta: DefinitionMeta = {battery: {}};
+        if (args.voltageToPercentage) meta.battery.voltageToPercentage = args.voltageToPercentage;
+        if (args.dontDividePercentage) meta.battery.dontDividePercentage = args.dontDividePercentage;
+        result.meta = meta;
+    }
+
+    return result;
+}
+
+export function deviceTemperature(args?: Partial<NumericArgs>) {
+    return numeric({
+        name: 'device_temperature',
+        cluster: 'genDeviceTempCfg',
+        attribute: 'currentTemperature',
+        reporting: {min: '5_MINUTES', max: '1_HOUR', change: 1},
+        description: 'Temperature of the device',
+        unit: '°C',
+        access: 'STATE_GET',
+        entityCategory: 'diagnostic',
+        ...args,
+    });
+}
+
+export function identify(args?: {isSleepy: boolean}): ModernExtend {
+    args = {isSleepy: false, ...args};
+    const normal: Expose = e.enum('identify', ea.SET, ['identify']).withDescription('Initiate device identification').withCategory('config');
+    const sleepy: Expose = e.enum('identify_sleepy', ea.SET, ['identify_sleepy'])
```

Comment:
I think only the description should vary for sleepy devices, not the key (`identify_sleepy`)

Reply:
I added it to better indicate that it is sleepy in external systems (which don't have support for descriptions). But maybe you are right.